### PR TITLE
Single stack

### PR DESCRIPTION
**Pull request by tomrf1**
Consolidate the lambdas into a single stack
This makes it easier to share code, and also share the api gateway

It means:
- 1 teamcity build
- 1 riffraff deploy
- 1 deployable .zip with all of the code
- 3 lambdas (eventually) all using the same .zip

### CODE REVIEW

File: src/braze-upload/lambda/lambda.ts [hunk 1]
```diff
@@ -56,7 +56,8 @@ export async function handler(event: Event, context: any): Promise<any> {
 
     const resultPromises = maybeReferralCodes
         .filter(maybeReferralCode => !!maybeReferralCode)
-        .map(async referralCode => {
+        .map(c => c as string)  // typescript doesn't know that the above line filters to strings only
```
Comment (paulbrown1982) [inline src/braze-upload/lambda/lambda.ts:59]: You could delete this line if you make the above line:
```suggestion
        filter(maybeReferralCode => String.apply(!!maybeReferralCode))
```

Comment (tomrf1) [reply src/braze-upload/lambda/lambda.ts:59]: unfortunately typescript still considers that a `string | null` without the cast

Comment (tomrf1) [reply src/braze-upload/lambda/lambda.ts:59]: I think the weakness is in the `filter` definition, where the return type is the same:
`filter(callbackfn: (value: T, index: number, array: T[]) => any, thisArg?: any): T[];`

I've been meaning to try the fp-ts library (which seems quite scala-influenced), which has an [Array type](https://gcanti.github.io/fp-ts/modules/Array.ts.html) which does what we need:
`const filterMap: <A, B>(f: (a: A) => Option<B>) => (fa: A[]) => B[]`
I'll leave that for a future refactor...

Comment (paulbrown1982) [reply src/braze-upload/lambda/lambda.ts:59]: That's weird the `| null`, given this:

Screen Shot 2020-09-24 at 08 21 36

**Review by paulbrown1982 — APPROVED** [review]
👍  LGTM

File: src/braze-upload/lambda/lambda.ts (hunk 1 above)
Comment (tomrf1) [reply src/braze-upload/lambda/lambda.ts:59]: The original array has type `(string | null)[]`, and filter doesn't allow you to change the type to `string[]` - there has to be a `map` as well for that

Comment (paulbrown1982) [reply src/braze-upload/lambda/lambda.ts:59]: Gotcha. Using `.map` is better.

**Pull request merged**